python/spglib/spglib.py

```diff
@@ -1215,28 +1241,76 @@ def get_magnetic_spacegroup_type(uni_number) -> dict | None:
     """
     _set_no_error()
 
-    keys = (
-        "uni_number",
-        "litvin_number",
-        "bns_number",
-        "og_number",
-        "number",
-        "type",
-    )
     msg_type_list = _spglib.magnetic_spacegroup_type(uni_number)
-    _set_error_message()
 
     if msg_type_list is not None:
-        msg_type = dict(zip(keys, msg_type_list))
-        for key in msg_type:
-            if key not in ["uni_number", "litvin_number", "number", "type"]:
-                msg_type[key] = msg_type[key].strip()
+        msg_type = MagneticSpaceGroupType(
+            uni_number=msg_type_list[0],
+            litvin_number=msg_type_list[1],
+            bns_number=msg_type_list[2].strip(),
+            og_number=msg_type_list[3].strip(),
+            number=msg_type_list[4],
+            type=msg_type_list[5],
+        )
         return msg_type
     else:
+        _set_error_message()
         return None
 
 
-def get_pointgroup(rotations):
+def get_magnetic_spacegroup_type_from_symmetry(
+    rotations: np.ndarray,
+    translations: np.ndarray,
+    time_reversals: np.ndarray,
```

Comment:
I didn't look into it, what's the difference between `np.ndarray` and `NDarray[float]`? At the very least it would be nice to include the data-type in it.